stdlib/universe/chande_momentum_oscillator.go

```diff
@@ -0,0 +1,307 @@
+	return plan.NarrowTransformationTriggerSpec{}
+}
+
+func createChandeMomentumOscillatorTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*ChandeMomentumOscillatorProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewChandeMomentumOscillatorTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type chandeMomentumOscillatorTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	n       int64
+	columns []string
+}
+
+func NewChandeMomentumOscillatorTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *ChandeMomentumOscillatorProcedureSpec) *chandeMomentumOscillatorTransformation {
+	return &chandeMomentumOscillatorTransformation{
+		d:       d,
+		cache:   cache,
+		n:       spec.N,
+		columns: spec.Columns,
+	}
+}
+
+func (t *chandeMomentumOscillatorTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *chandeMomentumOscillatorTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("chande momentum oscillator found duplicate table with key: %v", tbl.Key())
+	}
+	cols := tbl.Cols()
+	doChandeMomentumOscillator := make([]bool, len(cols))
+	for j, c := range cols {
+		found := false
+		for _, label := range t.columns {
+			if c.Label == label {
+				if c.Type != flux.TInt && c.Type != flux.TUInt && c.Type != flux.TFloat {
+					return fmt.Errorf("cannot take chande momentum oscillator of column %s (type %s)", c.Label, c.Type.String())
+				}
+				found = true
+				break
+			}
+		}
+
+		if found {
+			mac := c
+			mac.Type = flux.TFloat
+			_, err := builder.AddCol(mac)
+			if err != nil {
+				return err
+			}
+			doChandeMomentumOscillator[j] = true
+		} else {
+			_, err := builder.AddCol(c)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return tbl.Do(func(cr flux.ColReader) error {
+		if cr.Len() == 0 || len(cr.Cols()) == 0 {
+			return nil
+		}
+
+		for j, c := range cr.Cols() {
+			if !doChandeMomentumOscillator[j] {
+				for i := int(t.n); i < cr.Len(); i++ {
+					if err := builder.AppendValue(j, execute.ValueForRow(cr, i, j)); err != nil {
+						return err
+					}
+				}
+				continue
+			}
+			if err := t.do(int(t.n), cr, c, builder, j); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func (t *chandeMomentumOscillatorTransformation) do(n int, cr flux.ColReader, c flux.ColMeta, builder execute.TableBuilder, j int) error {
+	var sumUp float64
+	var sumDown float64
+	sumUp = 0
+	sumDown = 0
+
+	switch c.Type {
+	case flux.TInt:
+		arrValues := cr.Ints(j)
```

Comment:
it looks to me like all of the values in this case besides `curr` and `prev` are float types.  Since  `float(curr) - float(prev) == float(curr - prev)`   it seems like you can move most of this logic into a function   `func nextCMO(curr, prev float) (float,float)`   that you call like,   `val, diffNAgo = nextCMO(float64(curr), float64(prev))`  

Then all of these cases become much shorter, easier to read.